modules/extending_and_customizing.md

```diff
@@ -0,0 +1,155 @@
+
+### Custom cli
+
+Standalone blueprints can be executed using `yo`, but `yo` is aggressive in generators discovery (can be slow) and lacks some improvements JHipster cli provides like help and jhipster integration.
+So we recommend using `jhipster` cli or custom cli based on generator-jhipster.
+
+`jhipster` command executes the generator-jhipster version you have installed globally.
+A custom cli will execute the dependent generator-jhipster version, so will make sure to use the supported generator-jhipster version.
+
+Custom cli allows to execute a custom generator and will be covered at [Creating a module](/modules/creating-a-module).
+
+## Development
+
+### Application configuration:
+
+JHipster configuration follows [yeoman configuration](https://yeoman.io/authoring/storage.html) pattern and provides additional support for blueprint config.
+
+`config` and `jhipsterConfig` properties stores the common config and write to `generator-jhipster` key at `.yo-rc.json` file.
+`blueprintStorage` and `blueprintConfig` properties stores the blueprint specific config and write to `generator-jhipster-(my-blueprint)` key at `.yo-rc.json` file.
+
+`config` and `blueprintStorage` are [Storage instances](https://yeoman.github.io/generator/Storage.html).
+While `jhipsterConfig` and `blueprintConfig` are [proxy objects](https://yeoman.github.io/generator/Storage.html#createProxy) for `config` and `blueprintStorage` storages for convenience.
+
+### Global constants:
+
+You can use constants in [generator-constants](https://github.com/jhipster/generator-jhipster/blob/main/generators/generator-constants.js):
+
+```javascript
+    const javaDir = `${jhipsterConstants.SERVER_MAIN_SRC_DIR + this.packageFolder}/`;
+    const resourceDir = jhipsterConstants.SERVER_MAIN_RES_DIR;
+    const webappDir = jhipsterConstants.CLIENT_MAIN_SRC_DIR;
+```
+
+### Functions:
+
+You can use all functions in [generator-base](https://github.com/jhipster/generator-jhipster/blob/main/generators/generator-base.js):
+
+## Running local Blueprint version for development
+
+During development of blueprint, please note the below steps. they are very important.
+
+1. Link your blueprint globally 
+
+Note: If you do not want to link the blueprint(step 3) to each project being created.
+
+```bash
+cd generator-jhipster-my-blueprint
+npm link
+```
+
+2. Link a development version of JHipster to your blueprint (optional: required only if you want to use a non-released JHipster version, like the main branch or your own custom fork)
+
+```bash
+cd generator-jhipster
+npm link
+
+cd generator-jhipster-my-blueprint
+npm link generator-jhipster
+```
+
+Or install generator-jhipster from git:
+
+```bash
+cd generator-jhipster-my-blueprint
+npm install jhipster/generator-jhipster
+```
+
+3. Create a new folder for the app to be generated, and run JHipster ignoring JHipster dependencies (otherwise a released version will be installed each time npm install/ci is called)
```

Comment:
```suggestion
3. Create a new folder for the app to be generated, and run JHipster ignoring JHipster dependencies (otherwise a released version will be installed each time `npm install/ci` is called)
```